File: project/backend/springboot/src/core/src/main/java/mvm/micro/core/application/usecase/HealthCheck/HealthCheckQueryHandler.java
package mvm.micro.core.application.usecase.HealthCheck;

import mvm.micro.shared.domain.bus.query.QueryHandler;
import mvm.micro.shared.infrastructure.BaseResponse;
import org.springframework.stereotype.Service;
import java.util.HashMap;
import java.util.Map;

@Service("HealthCheckQueryHandler")
public class HealthCheckQueryHandler implements QueryHandler<HealthCheckQuery, BaseResponse> {
    @Override
    public BaseResponse handle(HealthCheckQuery query) {
        Map<String, Object> initialData = new HashMap<>();
        initialData.put("status", "ok");

        return new BaseResponse(new HashMap<>(initialData));
    }
}

File: project/backend/django/myproject/core/application/UseCase/HealthCheck/health_check_query_handler.py
from myproject.shared.domain.bus.query.query_handler import QueryHandler
from myproject.shared.domain.response import BaseResponse
from myproject.core.application.UseCase.HealthCheck.health_check_query import HealthCheckQuery

class HealthCheckQueryHandler(QueryHandler):
    def handle(self, query: HealthCheckQuery):
        return BaseResponse(
            data={"status": "OK"},
            message="Health check successful",
            status=200
        ).to_dict()
